src/events/interactionCreate.ts

```diff
@@ -20,8 +28,18 @@ export const interactionCreate = onEvent('interactionCreate', {
 	once: false,
 	async execute(interaction) {
 		try {
+			// Don't respond to bots or ourselves
+			if (interaction.user.bot) return;
+			if (interaction.user.id === interaction.client.user.id) return;
+
 			if (interaction.isCommand()) {
-				await handleInteraction(interaction);
+				const context = await handleInteraction(interaction);
+				await handleCommandInteraction(context, interaction);
+			}
+
+			if (interaction.isButton()) {
+				const context = await handleInteraction(interaction);
```

Comment:
These 2 context-generation lines can probably be moved out of the if-statements so they aren't duplicated